daffodil-lib/src/main/scala/org/apache/daffodil/schema/annotation/props/PropertyScoping.scala

```diff
@@ -120,7 +127,7 @@ trait FindPropertyMixin extends PropTypes {
   
   val propCache = new scala.collection.mutable.LinkedHashMap[String, PropertyLookupResult]
 
-  def findPropertyOption(pname: String): PropertyLookupResult = {
+  def findPropertyOption(pname: String, isForExpression: Boolean = false): PropertyLookupResult = {
```

Comment:
Add scaladoc explaining purpose of isForExpression and that only for properties that can be expressions have to pass true. 

Maybe the name should be "canBeExpressionOrNot" because many properties are optionally for expressions e.g, dfdl:byteOrder which can be literally "bigEndian" or can be an expression. Other properties like dfdl:inputValueCalc are always an expression, but we don't need this to be passed as true for them.

Reply:
Added a scaladoc. The autogenerated block that pops up in IntelliJ for the function header didn't seem to match other functions in the file. Does this scaladoc look ook?

Reply:
Looks good.